### PR TITLE
NRDP flexible opts

### PR DESCRIPTION
**Pull request by ecklm**
Icinga2 uses a different manner of command configuration and it does not support specific parameter ordering.

This patch(set) is required to solve this problem by making option processing flexible/order-independent.

### CODE REVIEW

**Comment by ecklm** [comment]
I know that this is already conflicting with PR #4 , I'll rebase this one once you accept the other.

**Comment by ecklm** [comment]
@dennyzhang until you tell your opinion about the patch I already submitted, I won't start updating the rest of the nrdp bash scripts as I only needed this one. Afterwards, if I manage to find the time, I'm glad to do so.

**Comment by dennyzhang** [comment]
@ecklm , thanks for working on this incompatibility issue of Icinga2.

Could you help to validate the change works for both old nagios and new icinga?

**Comment by ecklm** [comment]
In general, this modification is independent of icinga2 or nagios itself, it just changes the processing of command line parameters so that it does not require a fixed order of them.

But to prove, NRDP configs use this command in the fixed order of `-w -c` and some of the pid specification commands. This order still works, therefore it should not interfere with the old configs.

```bash
$ ./check_proc_mem.sh -w 3 -c 4 --cmdpattern ssh
Memory: CRITICAL RES: 5 MB - VIRT: 12 MB used!|RES=5242880;;;;

$ ./check_proc_mem.sh -w 3 -c 7 --pid 744
Memory: WARNING RES: 5 MB - VIRT: 12 MB used!|RES=5242880;;;;

$ ./check_proc_mem.sh -w 3 -c 7 --pidfile /run/sshd.pid
Memory: WARNING RES: 5 MB - VIRT: 12 MB used!|RES=5242880;;;;
```

The difference is that it now **also** works this way:
```bash
$ ./check_proc_mem.sh --pidfile /run/sshd.pid -w 3 -c 7 
Memory: WARNING RES: 5 MB - VIRT: 12 MB used!|RES=5242880;;;;
```

**Comment by dennyzhang** [comment]
@ecklm , thanks for the clarification. Make sense to me.

**Comment by ecklm** [comment]
Okay. During the next weeks, I'll try to update the remaining shell scripts, too.

**Comment by ecklm** [comment]
Need fix:
- [x] check_proc_threadcount.sh
- [x] check_proc_cpu.sh
- [x] check_proc_mem.sh
- [x] check_proc_fd.sh

The following do not need fix:

- check_disk_size.sh

**Comment by ecklm** [comment]
I consider this patchset final

File: nagios_plugins/check_proc_cpu/check_proc_cpu.sh [hunk 1]
```diff
@@ -61,6 +28,63 @@ else
...
+            exit 3
+        ;;
+    esac
+done
+
+num_re='^[0-9]+$'
+if ! [[ "$warn_cpu" =~ $num_re ]] || ! [[ "$criti_cpu" =~ $num_re ]]
+then
+    echo "ERROR: Warning or Critical level is not a number"
+	exit 3
```
Comment (dennyzhang) [inline nagios_plugins/check_proc_cpu/check_proc_cpu.sh]: Would you like to fix the indentation?

File: nagios_plugins/check_proc_cpu/check_proc_cpu.sh [hunk 2]
```diff
@@ -61,6 +28,63 @@ else
...
+
+if [ -z "$pid" ]; then
+    echo "ERROR: no related process is found"
+    exit 2
+fi
+
+cpuUsage=$(ps -p "$pid" -o "%cpu" | tail -n 1 | sed -e 's/^[ \t]*//')
+
+if [ "$(echo "$cpuUsage>$criti_cpu" | bc)" = "1" ]; then
+    echo "Critical CPU used by process($pid) is $cpuUsage % is more than $criti_cpu %!|CPU=$cpuUsage"
```
Comment (dennyzhang) [inline nagios_plugins/check_proc_cpu/check_proc_cpu.sh]: The log message doesn't look like a valid english sentence. Could you like to polish it a little bit? Same for L84

File: nagios_plugins/check_proc_cpu/check_proc_cpu.sh (hunk 1 above)
Comment (ecklm) [reply nagios_plugins/check_proc_cpu/check_proc_cpu.sh]: Oh, I thought I already fixed it. I will.

File: nagios_plugins/check_proc_cpu/check_proc_cpu.sh (hunk 2 above)
Comment (ecklm) [reply nagios_plugins/check_proc_cpu/check_proc_cpu.sh]: Done. By the way, originally I didn't change the error messages.

File: nagios_plugins/check_proc_cpu/check_proc_cpu.sh (hunk 1 above)
Comment (ecklm) [reply nagios_plugins/check_proc_cpu/check_proc_cpu.sh]: I fixt the rest of them, too.

**Pull request merged**